src/main/scala/uk/gov/nationalarchives/dp/client/EntityClient.scala

```diff
@@ -335,6 +372,26 @@ object EntityClient {
         }
       } yield body
     }
+
+    override def updateIdentifiers(entity: Entity, identifiers: Seq[IdentifierResponse]): F[Seq[IdentifierResponse]] = {
+      identifiers.map { identifier =>
+        val xml = <Identifier xmlns="http://preservica.com/XIP/v6.9">
```

Comment:
They're not quite the same. The one here is an `IdentifierResponse` which includes the id of the identifier as this is needed to update them. We could change it so there's a common trait but it seems like overkill.

Reply:
But couldn't this method/function just take the `identifierName` and `value` instead of the Object (since your not using `id` to generate it)? So

```
def methodName(identifierName: String, value: String ) = {
          identifierAsXml: String = {
          val xml = <Identifier xmlns="http://preservica.com/XIP/v6.5">
            <Type>{identifier.identifierName}</Type>
            <Value>{identifier.value}</Value>
          </Identifier>
          new PrettyPrinter(80, 2).format(xml)
        }
       s"""<?xml version="1.0" encoding="UTF-8" standalone="yes"?>\n$identifierAsXml"""
}
```

So in the `identifiers.map`, you just do `methodName(identifier.identifierName, identifier.value)`

Reply:
That's a good point yeah. Ok, I've moved it to a private method :+1: